### PR TITLE
Adds --keep-context flag for 'minikube start'

### PR DESCRIPTION
**Pull request by gauthierbl**
This PR adds a new flag to `minikube start` that will keep the existing kubectl context.

Ref #687

### CODE REVIEW

**Comment by gauthierbl** [comment]
I signed it!

**Comment by dlorenc** [comment]
@minikube-bot ok to test

**Comment by dlorenc** [comment]
Sorry for the delay! Most of the minikube maintainers were on vacation for the holidays. We'll take a look at this today.

**Comment by gauthierbl** [comment]
@dlorenc no problem, and I understand, hope everyone had a good holiday.

As for the failing tests, any idea what might be causing those? Is there a way to run those tests locally so I can see what is going on?

**Comment by dlorenc** [comment]
@gauthierbl these tests run across different platforms, you can run them locally but it's a bit tricky unless you have physical linux,mac and windows machines.

@r2d4 is looking into the flakes, they shouldn't block this from going in.

File: cmd/minikube/cmd/start.go [hunk 1]
```diff
@@ -136,15 +137,21 @@ func runStart(cmd *cobra.Command, args []string) {
...
 	clientKey := constants.MakeMiniPath("apiserver.key")
-	if err := setupKubeconfig(name, kubeHost, certAuth, clientCert, clientKey); err != nil {
+	if err := setupKubeconfig(name, kubeHost, certAuth, clientCert, clientKey, keepContext); err != nil {
 		glog.Errorln("Error setting up kubeconfig: ", err)
 		cmdUtil.MaybeReportErrorAndExit(err)
 	}
-	fmt.Println("Kubectl is now configured to use the cluster.")
+
+	if keepContext {
+		fmt.Println("The local Kubernetes cluster has started. The kubectl context has not been alterted, kubectl will require --context or --kubeconfig to use the local Kubernetes cluster.")
```
Comment (gauthierbl) [inline cmd/minikube/cmd/start.go]: I don't think --kubeconfig could be used unless the user did something like create a minikube context in a different kubeconfig. That seems ... unusual. If they did that then I guess they would know what they did and what they needed to do.

I'll remove the kubeconfig suggestion and change the wording on --context.

Comment (gauthierbl) [inline cmd/minikube/cmd/start.go]: done and squashed.

**Comment by dlorenc** [comment]
Thanks!

**Pull request merged**